source/contact.ts

```diff
@@ -12,7 +11,7 @@ export class Contact implements IContact {
 
 	#avatar: string;
 
-	public constructor({ id, name, starred }: IContactData, profile: IProfile) {
+	public constructor({ id, name, starred }: Omit<IContactData, "addresses">, profile: IProfile) {
```

Comment:
I think using the following `validate` method internally, would help us validate the entire contact object and make sure address data would not be undefined, in case the consumer tries to add corrupted address objects when calling `create` or `update`.  What do you think?

```suggestion
	private validate(data: IContactData): IContactData {
		const { error, value } = Joi.object({
			id: Joi.string().required(),
			name: Joi.string().required(),
			addresses: Joi.array()
				.min(1)
				.items(
					Joi.object({
						coin: Joi.string().required(),
						network: Joi.string().required(),
						address: Joi.string().required(),
					}),
				),
			starred: Joi.boolean().required(),
		}).validate(data, { stripUnknown: true, allowUnknown: true });

		if (error !== undefined) {
			throw error;
		}

		return value as IContactData;
	}

	public constructor({ id, name, starred }: Omit<IContactData, "addresses">, profile: IProfile) {
```